core/src/context.rs

```diff
@@ -23,11 +30,32 @@ impl Context {
         logger: Arc<Mutex<Logger>>,
         persister: Arc<Mutex<Persister>>,
     ) -> Context {
+        let (tx_action, _) = sync_channel(100);
+        let (tx_observer, _) = sync_channel(100);
```

Comment:
Magic number: Please use a global/struct const instead and possibly add a comment explaining the rationale behind that number.